packages/dd-trace/src/proxy.js

```diff
@@ -33,6 +27,20 @@ class LazyModule {
   }
 }
 
+function lazyProxy (obj, property, getClass, ...args) {
```

Comment:
why not just...

```js
function lazyProxy(obj, property, getClass, ...args) {
  const configurable = true
  const writable = true
  const enumerable = true
  Reflect.defineProperty(obj, property, {
    get () {
    	const RealClass = getClass()
    	const value = new RealClass(...args)
    	Reflect.defineProperty(obj, property, {
    		value,
    		configurable,
    		writable,
    		enumerable
    	})
    	return value
    },
    configurable,
    writable,
    enumerable
  })
}

Reply:
There are scenarios where this doesn't work, for example any property defined in the constructor. It's also unclear that this is much faster as we've already established in the past that descriptors can be pretty slow.

Reply:
Oh wait I think I misunderstood what you're doing, basically you're loading when the property from the tracer is read, not the property of the instance. Maybe that makes more sense, let me think about it.

@simon-id I took the proxy idea from your draft PR for AppSec, what was the reasoning for using a Proxy?

Reply:
Updated with a property descriptor instead of a proxy.

Reply:
Eh no reason except it's less verbose, the proxy destructs itself after first use anyway. Property descriptor should work just the same i assume

Reply:
@bengl Is the new code fine?

Reply:
@bengl Any remaining concerns?